sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -145,10 +148,23 @@ static void initializeIntegrationsAndProcessors(
     options.addEventProcessor(new ViewHierarchyEventProcessor(options));
     options.addEventProcessor(new AnrV2EventProcessor(context, options, buildInfoProvider));
     options.setTransportGate(new AndroidTransportGate(options));
-    final SentryFrameMetricsCollector frameMetricsCollector =
-        new SentryFrameMetricsCollector(context, options, buildInfoProvider);
-    options.setTransactionProfiler(
-        new AndroidTransactionProfiler(context, options, buildInfoProvider, frameMetricsCollector));
+
+    // Check if the profiler was already instantiated in the startup.
+    // We use the Android profiler, that uses a global start/stop api, so we need to preserve the
+    // state of the profiler, and it's only possible retaining the instance.
+    final @Nullable ITransactionProfiler startupProfiler =
+        AppStartMetrics.getInstance().getStartupProfiler();
+    if (startupProfiler != null) {
```

Comment:
dunno if we need synchronization here, in theory `SentryAndroid.init` might be called from a background thread, but the `shutdown` from the contentprovider might be called in the meantime